### DISCUSSION TITLE
Approach to implement RCCC camera?

### DISCUSSION THREAD

**Discussion by roque2205** [carla-simulator/carla#4139] [Q&A] [answered]
Hi, I'm trying to introduce an RCCC (red-clear-clear-clear) camera, which is similar to the RGB camera but with a pixel-wise filter. It should convert the image to grayscale, but in the way that in each array of 2x2 pixels, the lower right pixel's intensity is set to the R value, whereas the other pixels' intensities are calculated by a weighted average of R, G and B. I thought I could manage this by adapting the imageserializer class. As this one is setting alpha values of the image, I figured it should be possible to set R, G and B in the same way to create a grayscale picture. This new RCCC serializer is paired with a new RCCC camera class which I created similarly to the RGB camera. Everything works really fine except for that the resulting images from the RCCC camera look exactly like the ones from the RGB camera. I would really, really appreciate hints why this doesn't work the way I expected it and how and where to best implement this filter.

**Comment by roque2205** [comment, marked as answer]
I solved it myself using a postprocessing material. Message me if you need advice.

**Reply by Cenbylin** [reply]
Hi roque,

Did you use your own postprocessing material? I would like to ask if can I do this without the need to recompile the whole Carla. 

Thanks for your advice.

**Reply by roque2205** [reply]
Hi Cenbylin, I used my own postprocessing material and created an own c++ class for this rccc camera, so I'm afraid you would have to recompile Carla.
